src/Compilers/Core/Portable/MetadataReader/MetadataDecoder.cs

```diff
@@ -1844,60 +1869,17 @@ protected TypeSymbol DecodeFieldSignature(ref BlobReader signatureReader, out bo
 
             try
             {
-                // See if there is a Volatile modifier.
+                var isVolatileFound = false;
                 SignatureTypeCode typeCode;
-                ArrayBuilder<ModifierInfo<TypeSymbol>> customModifierBuilder = null;
```

Comment:
Will try to introduce in the next iteration a simpler version that doesn't use capturing lambdas. I don't like to duplicate the same logic in 3 places (DecodeModifiers, DecodeParameters, DecodeFields).

---
In reply to: [118054960](https://github.com/dotnet/roslyn/pull/19658#discussion_r118054960) [](ancestors = 118054960)